jenkins/scality-glance-store-functional-tests/30-run-tests.sh

```diff
@@ -1,26 +1,26 @@
 #!/bin/bash -xue
 
-TEMPEST_DIR=/opt/stack/tempest
-
-virtualenv venv_for_nosetests
-set +o nounset # See https://github.com/pypa/virtualenv/issues/150
-source venv_for_nosetests/bin/activate
-set -o nounset
-
-pip install -r $TEMPEST_DIR/test-requirements.txt -r $TEMPEST_DIR/requirements.txt nose
-
+cd /opt/stack/tempest
 set +e
-nosetests -w $TEMPEST_DIR/tempest/api/image --exe --with-xunit --xunit-file=${WORKSPACE}/nosetests.xml
+tox -e all -- tempest.api.image
+RC=$?
+testr last --subunit > $WORKSPACE/testrepository.subunit
 set -e
 
-set +o nounset # See https://github.com/pypa/virtualenv/issues/150
-deactivate
-set -o nounset
-
 echo "Entering WORKSPACE."
 cd $WORKSPACE
 mkdir jenkins-logs
 echo "Creating jenkins-log directory."
 cp -R /opt/stack/logs/* jenkins-logs/
 sudo chown jenkins jenkins-logs/*
-exit 0
+
+# Create a test result report
+set +e
+if sudo pip install junitxml; then
```

Comment:
You should install python-subunit instead, AFAIK junitxml is deprecated.
Plus I think you should delete the if clause and the 'touch': if the package fails to install, you will have zero informations on which tests succeed or not so the job should be marked as 'failed'.
This if clause probably does not achieve the desired effect anyway since we use -e.